### PR TITLE
Add tests for hisat2

### PR DESCRIPTION
**Pull request by ntoda03**
## PR checklist

Closes #XXX 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `<SOFTWARE>.version.txt` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Review by grst — DISMISSED** [review]
Can you please update this to use the new config file for test data?

**Comment by ntoda03** [comment]
Ok, it's been updated to use the new test config file.

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Some `md5` for `bam` and `ht2` checks are failing since they include variable stuff inside (command used for instance).  Just check for them to exist, see [here](https://pytest-workflow.readthedocs.io/en/stable/#test-options).

**Comment by ntoda03** [comment]
> Some `md5` for `bam` and `ht2` checks are failing since they include variable stuff inside (command used for instance). Just check for them to exist, see [here](https://pytest-workflow.readthedocs.io/en/stable/#test-options).

Ok I got rid of the md5s for the bam files. For the ht2 checks I just had to update them with the new build method.

**Review by JoseEspinosa — APPROVED** [review]
LGTM

File: software/hisat2/align/meta.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          respectively.
+  - index:
+      type: file
+      description: HISAT2 genome index file
+      pattern: "*.ht2"
+  - splicesites:
+      type: file
+      description: Splices sites in gtf file
+      pattern: "*.{txt}"
+
```
Comment (drpatelh) [inline software/hisat2/align/meta.yml:36]:
```suggestion
```

File: software/hisat2/align/meta.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      pattern: "*.{bam}"
+  - summary:
+      type: file
+      description: Aligment log
+      pattern: "*.log"
+  - version:
+      type: file
+      description: File containing software version
+      pattern: "*.{version.txt}"
+
```
Comment (drpatelh) [inline software/hisat2/align/meta.yml:56]:
```suggestion
```

File: software/hisat2/extractsplicesites/meta.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+output:
+  - version:
+      type: file
+      description: File containing software version
+      pattern: "*.{version.txt}"
+  - splicesites:
+      type: file
+      description: Splices sites in gtf file
+      pattern: "*.{splice_sites.txt}"
+
```
Comment (drpatelh) [inline software/hisat2/extractsplicesites/meta.yml:32]:
```suggestion
```

**Review by drpatelh — APPROVED** [review]
Thank you for doing this @ntoda03 ! 😎

**Pull request merged**